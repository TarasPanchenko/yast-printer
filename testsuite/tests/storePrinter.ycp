/**
 * File:
 *  storePrinter.ycp
 *
 * Module:
 *  Printer configurator
 *
 * Summary:
 *  Tests for basic translation functions.
 *
 * Authors:
 *  Petr Blahos <blahos@suse.cz>, 2001
 *
 * $Id$
 *
 */


{
    import "Pkg";

    include "testsuite.ycp";
    import "Printer";

    Printer::default = "";
    Printer::printcap = [];
    Printer::index = -1;
    Printer::printer = $[
	"name" : "lp",
	"uri" : "parallel:/dev/lp0",
	];
    Printer::storePrinter ();
    DUMP (Printer::printcap);
    DUMP (Printer::default);

    Printer::default = "";
    Printer::printcap = [];
    Printer::index = -1;
    Printer::printer = $[
	"name" : "zzz",
	"uri" : "lpd://unknown.suse.de/lp_rem",
	"database" : "foomatic",
    ];
    Printer::storePrinter ();
    DUMP (Printer::printcap);
    DUMP (Printer::default);

    Printer::index = -1;
    Printer::default = "";
    Printer::printer = $[
	"name" : "lp",
	"uri" : "filtering://localhost/zzz",
	"database" : "foomatic",
	];
    Printer::storePrinter ();
    DUMP (Printer::printcap);
    DUMP (Printer::default);

    Printer::default = "";
    Printer::index = 0;
    Printer::printer = $[
	"name" : "zzz",
	"uri" : "lpd://known.suse.de/lp",
	"database" : "foomatic",
	];
    Printer::storePrinter ();
    DUMP (Printer::printcap);
    DUMP (Printer::default);
}
