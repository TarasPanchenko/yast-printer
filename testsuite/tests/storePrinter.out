Return	nil
Dump	[$["cups-state":"void", "lprng-state":"void", "name":"lp", "uri":"parallel:/dev/lp0"]]
Dump	lp
Return	nil
Dump	[$["cups-state":"changed", "database":"foomatic", "lprng-state":"changed", "name":"zzz", "uri":"lpd://unknown.suse.de/lp_rem"]]
Dump	zzz
Read	.target.dir "/usr/lib/cups/backend" []
Return	nil
Dump	[$["cups-state":"changed", "database":"foomatic", "lprng-state":"changed", "name":"zzz", "uri":"lpd://unknown.suse.de/lp_rem"], $["cups-state":"void", "database":"foomatic", "lprng-state":"void", "name":"lp", "uri":"filtering://localhost/zzz"]]
Dump	lp
Return	nil
Dump	[$["cups-state":"changed", "database":"foomatic", "lprng-state":"changed", "name":"zzz", "uri":"lpd://known.suse.de/lp"], $["cups-state":"void", "database":"foomatic", "lprng-state":"void", "name":"lp", "uri":"filtering://localhost/zzz"]]
Dump	zzz
